### PR TITLE
feat: Add delete highlights implementation

### LINKED ISSUE #906: Feature: add deleted button to the highlights
### Type of feature

🍕 Feature

### Current behavior

currently there's no delete button to delete highlight
image

### Suggested solution

_No response_

### Additional context

https://insights.opensauced.pizza/user/dev-phantom

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by OgDev-01**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR:

- Adds delete button to delete a highlight
- Adds alert dialog to prompt a user to confirm delete action or cancel

## Related Tickets & Documents
Fixes #906

## Mobile & Desktop Screenshots/Recordings
image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by brandonroberts — APPROVED** [review]
Functionally works. List is not refreshed, but may be covered in #957

**Review by bdougie — CHANGES_REQUESTED** [review]
Missed this in the design review, but can we light the delete button to match what is on the inisights page. [Link to Figma](https://www.figma.com/file/OpVX6WT7dmWqnwRuEvADMF/OpenSauced-%E2%80%94-Insights-(insights.opensauced.pizza)?node-id=7217-9531&t=TFWA4W254HfVIBvi-0)

Screen Shot 2023-03-08 at 9 07 57 AM


cc @getaheaddev 

update: on hover is not great.
Screen Shot 2023-03-08 at 10 28 07 AM

File: lib/hooks/useFetchUserHighlights.ts [hunk 1]
```diff
@@ -13,16 +13,15 @@ const useFetchUserHighlights = (username: string) => {
   const pageQuery = page ? `page=${page}` : "";
   const limitQuery = limit ? `&limit=${limit}` : "";
```
Comment (Deadreyo) [inline lib/hooks/useFetchUserHighlights.ts:14]: If you deleted the usage of `pageQuery` & `limitQuery`, maybe you should delete their definitions as well, provided that we won't need them later on.

Comment (Deadreyo) [reply lib/hooks/useFetchUserHighlights.ts:14]: The states as well.

File: lib/hooks/deleteHighlight.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        Accept: "application/json",
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${sessionToken}`
+      },
+      method: "DELETE"
+    });
+
+    if (res.ok) {
+      return true;
+    }
```
Comment (Deadreyo) [inline lib/hooks/deleteHighlight.ts:19]: if res is not ok, what happens? Nothing is returned. IMO should throw an error.

**Review by Deadreyo — CHANGES_REQUESTED** [review]
Just some little details

File: lib/hooks/deleteHighlight.ts (hunk 2 above)
Comment (bdougie) [reply lib/hooks/deleteHighlight.ts:19]: We should catch the errors and display the error message.

File: lib/hooks/useFetchUserHighlights.ts [hunk 3]
```diff
@@ -13,8 +13,7 @@ const useFetchUserHighlights = (username: string) => {
...
   const { data, error } = useSWR<useFetchUserHighlightsResponse, Error>(
-    `users/${username}/highlights?${pageQuery}${limitQuery}`,
-    publicApiFetcher as Fetcher<useFetchUserHighlightsResponse, Error>
+    `users/${username}/highlights?${pageQuery}${limitQuery}`
```
Comment (brandonroberts) [inline lib/hooks/useFetchUserHighlights.ts:16]: This needs to use the `publicApiFetcher` to connect to the correct API

Comment (OgDev-01) [reply lib/hooks/useFetchUserHighlights.ts:16]: Good catch! 👍

File: lib/hooks/deleteHighlight.ts [hunk 4]
```diff
@@ -16,6 +16,13 @@ const deleteHighlight = async (id: string) => {
 
     if (res.ok) {
       return true;
+    } else {
+      const error = new Error("HttpError");
+
+      error.message = `${res.status} ${res.statusText}`;
+      error.stack = JSON.stringify(await res.json());
```
Comment (Deadreyo) [inline lib/hooks/deleteHighlight.ts:23]: Maybe we should start using `axios` to automatically throw errors on non-ok requests

**Review by Deadreyo — APPROVED** [review]
LGTM

**Comment by brandonroberts** [comment]
@OgDev-01 resolve merge conflicts and this is good to go

**Review by bdougie — APPROVED** [review]

**Pull request merged**